clients/java/client/src/main/java/org/camunda/bpm/client/impl/RequestExecutor.java

```diff
@@ -65,25 +66,25 @@ protected RequestExecutor(RequestInterceptorHandler requestInterceptorHandler, O
 
   protected <T> T postRequest(String resourceUrl, RequestDto requestDto, Class<T> responseClass) {
     ByteArrayEntity serializedRequest = serializeRequest(requestDto);
-    HttpUriRequest httpRequest = RequestBuilder.post(resourceUrl)
+    ClassicHttpRequest httpRequest = ClassicRequestBuilder.post(URI.create(resourceUrl).normalize())
```

Comment:
Without normalizing the URI the [PaExceptionIT#shouldThrowUnknownHttpError](https://github.com/camunda/camunda-bpm-platform/blob/master/clients/java/client/src/it/java/org/camunda/bpm/client/variable/pa/PaExceptionIT.java#L161) test fails because it receives a 404 instead a 405.

The test sends a `POST complete` request without `taskId`, so the url looks like this: http://localhost:48080/engine-rest/engine/another-engine/external-task//complete. 

HttpClient 4 [normalizes the URI](https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/execchain/ProtocolExec.java#L133). The RequestConfig [`normalizeUri`](https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/config/RequestConfig.html#isNormalizeUri()) is true by default.
HttpClient 5 doesn't normalize. I also didn't find a config like the `normalizeUri` in 4.

I decided to normalize the uri in the client code. For consistency I also added it in the `getRequest`.

Alternatively, we could:
- Only normalize for the `complete` API.
- Only normalize in the `postRequest`.
- Don't normalize, it's just an edge case. Fix rather the test.